libraries/abstractions/secure_sockets/cellular/iot_secure_sockets.c

```diff
@@ -1007,6 +1026,85 @@ static int32_t prvCheckSetSockOptParams( Socket_t xSocket,
 
 /*-----------------------------------------------------------*/
 
+#if ( ( CELLULAR_SUPPORT_GETHOSTBYNAME == 0 ) )
+
+    static uint32_t prvLookup( const char * pcHostName )
+    {
+        uint32_t resolvedAddress = 0, i = 0, foundInCache = 0;
+        uint32_t hostNameLength = strlen( pcHostName ); /* pcHostName is checked by caller. */
+        uint32_t cacheEntryLength = 0;
+
+        /* Ensure that the provided DNS name is of valid length. */
+        if( hostNameLength <= CELLULAR_IP_ADDRESS_MAX_SIZE )
+        {
+            /* Check if the provided DNS name already exists in cache. */
+            for( i = 0; i < GETHOSTBYNAME_CACHE_SIZE; i++ )
+            {
+                cacheEntryLength = strlen( _dnsCache[ i ] );
+
+                /* The length check is necessary to avoid false substring matches. */
+                if( ( cacheEntryLength == hostNameLength ) &&
+                    ( strncmp( _dnsCache[ i ], pcHostName, hostNameLength ) == 0 ) )
+                {
+                    foundInCache = 1;
+                    break;
+                }
+            }
+
+            if( foundInCache == 1 )
+            {
+                /* We return i + 1 to avoid returing 0 which is used to indicate a
+                 * lookup failure. */
+                resolvedAddress = i + 1;
+            }
+            else
+            {
+                /* Find an empty place in the cache and copy the DNS name. */
+                for( i = 0; i < GETHOSTBYNAME_CACHE_SIZE; i++ )
+                {
+                    if( _dnsCache[ i ][ 0 ] == '\0' )
+                    {
+                        strncpy( _dnsCache[ i ], pcHostName, CELLULAR_IP_ADDRESS_MAX_SIZE );
```

Comment:
That seems like unncessary copy of bytes after hostNameLength. Shouldn't it be:
```c
strncpy( _dnsCache[ i ], pcHostName, hostNameLength);
```

Reply:
This will be updated in a new PR as well.